card/mod/core/set/all/type.rb

```diff
@@ -18,6 +18,8 @@ def type_name
   type_card.try :name
 end
 
+alias_method :type, :type_name
```

Comment:
I would be in support of `type` returning a String and `type_name` returning a Card::Name.  That seems to clarify the issue for me.